atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransaction.java

```diff
@@ -1955,7 +1964,7 @@ private void commitWrites(TransactionService transactionService) {
                 // if there are no writes, we must still make sure the immutable timestamp lock is still valid,
                 // to ensure that sweep hasn't thoroughly deleted cells we tried to read
                 if (validationNecessaryForInvolvedTablesOnCommit()) {
-                    preCommitRequirementValidator.throwIfImmutableTsOrCommitLocksExpired(null);
+                    preCommitRequirementValidator.throwIfImmutableTsOrCommitLocksExpired(locksManager.getLockTokens());
```

Comment:
It is weird that we're specifying the `immutableTsLockToken` (via `locksManager.getLockTokens()`) here and `preCommitRequirementValidator` also holds a reference to the `immutableTsLockToken`. I agree strongly with https://github.com/palantir/atlasdb/pull/7352/files#r1801316887 - we should clean this up and have a single "lock manager per txn" class, similar to having a single "callback manager" class.

Reply:
For review: checked that `preCommitRequirementValidator.throwIfImmutableTsOrCommitLocksExpired` also validates the immutableTs lock is valid, so no change in behavior.